### PR TITLE
Added Documentation for Queue 

### PR DESCRIPTION
**Pull request by Saatvik21**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

`Yes`

### Description

I've added the docs for Queue Data Structure. 

`PS- I used C & C++ Code for queue using arrays as there was no code for Java/Python for queue using arrays.`

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes : #194

### CODE REVIEW

**Comment by Saatvik21** [comment]
@HarshCasper can you please review this PR, so that I can work on another issue.

File: docs/data-structures/queue.mdx [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+title: Queue
+slug: /data-structures/queue
+---
+
+## Introduction
+
+A ***queue*** is a logical data structure that works on the discipline ***FIFO(First-In-First-Out)***. It is an **ADT (Abstract Data Type)** where the insertion of elements occurs at one end, called **Rear**, and deletion occurs at another end, called **Front**. For example, people standing in a queue to avail some services or cars lined up in a queue are real-world examples of queue data structure.
+
+<p align="center">
+  <img width="540" height="400" src="https://user-images.githubusercontent.com/61618767/114686707-ac525500-9d30-11eb-882f-150835a03a58.gif" />
```
Comment (HarshCasper) [inline docs/data-structures/queue.mdx:11]: Did you create this GIF file?

File: docs/data-structures/queue.mdx [hunk 2]
```diff
@@ -0,0 +1,530 @@
+---
+title: Queue
+slug: /data-structures/queue
+---
+
+## Introduction
+
+A ***queue*** is a logical data structure that works on the discipline ***FIFO(First-In-First-Out)***. It is an **ADT (Abstract Data Type)** where the insertion of elements occurs at one end, called **Rear**, and deletion occurs at another end, called **Front**. For example, people standing in a queue to avail some services or cars lined up in a queue are real-world examples of queue data structure.
```
Comment (HarshCasper) [inline docs/data-structures/queue.mdx:8]: No need of unnecessary bolds. Remove them out from: 
- queue
- ADT 
- rear
- front

File: docs/data-structures/queue.mdx [hunk 3]
```diff
@@ -0,0 +1,530 @@
...
+
+Note that, in Python, we use **list** instead of an array for implementing queues.
+
+### Queue using Arrays
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+<Tabs
+  defaultValue="cpp"
```
Comment (HarshCasper) [inline docs/data-structures/queue.mdx:70]: Keep it C

File: docs/data-structures/queue.mdx [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+* `Dequeue` : Deletes an element from the **front** of the queue
+* `Front` : Returns the front element of the queue
+* `isEmpty` : Checks if a queue is empty or not
+* `Display` : Prints all the queue elements
+
+## Algorithm
+
+This section will discuss the algorithmic steps performed on the [basic operations](#basic-operations) of a queue.
+
+### Enqueue
```
Comment (HarshCasper) [inline docs/data-structures/queue.mdx:26]: Use Heading-4

File: docs/data-structures/queue.mdx [hunk 5]
```diff
@@ -0,0 +1,530 @@
...
+
+This section will discuss the algorithmic steps performed on the [basic operations](#basic-operations) of a queue.
+
+### Enqueue 
+
+* Check whether the queue is `full`.
+* If the queue is full, then display `"Queue is Full"` or `"OVERFLOW"` and terminate the function.
+* If the queue is not full, increment the `rear` pointer and `insert` the element.
+
+### Dequeue
```
Comment (HarshCasper) [inline docs/data-structures/queue.mdx:32]: Use Heading-4

File: docs/data-structures/queue.mdx [hunk 6]
```diff
@@ -0,0 +1,530 @@
...
+
+* Check whether the queue is `empty`.
+* If the queue is empty, then display `"Queue is Empty"` or `"UNDERFLOW"` and terminate the function.
+* If the queue is not empty, increment the `front` pointer and `delete` the element.
+
+### Front
```
Comment (HarshCasper) [inline docs/data-structures/queue.mdx:38]: Use Heading-4

File: docs/data-structures/queue.mdx [hunk 7]
```diff
@@ -0,0 +1,530 @@
...
+
+* Check whether the queue is empty.
+* If the queue is empty, then display `"Queue is Empty"` or `"UNDERFLOW"` and terminate the function.
+* If the queue is not empty, increment the `front` pointer and return the value of the element present.
+
+### isEmpty
```
Comment (HarshCasper) [inline docs/data-structures/queue.mdx:44]: Use Heading-4

File: docs/data-structures/queue.mdx [hunk 8]
```diff
@@ -0,0 +1,530 @@
...
+* If the queue is empty, then display `"Queue is Empty"` or `"UNDERFLOW"` and terminate the function.
+* If the queue is not empty, increment the `front` pointer and return the value of the element present.
+
+### isEmpty 
+
+* Check whether `front = -1 or front > rear`
+* If so, then return true.
+* Else, return false.
+
+### Display
```
Comment (HarshCasper) [inline docs/data-structures/queue.mdx:50]: Use Heading-4

File: docs/data-structures/queue.mdx [hunk 9]
```diff
@@ -0,0 +1,530 @@
...
+* Enqueue: `O(N)`
+* Dequeue: `O(1)` 
+* Display: `O(N)`
+
+## Credits
+
+* Queue using Arrays
+    * [Jivthesh M R](https://github.com/jivthesh) for the [C implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C/ds/Queueimplementation.c)
+    * [Pragya Jaiswal](https://github.com/pragyajaiswa05l) for the [C++ implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/ds/queue.cpp)
+
```
Comment (HarshCasper) [inline docs/data-structures/queue.mdx:527]: Remove the extra line

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/data-structures/queue.mdx (hunk 1 above)
Comment (Saatvik21) [reply docs/data-structures/queue.mdx:11]: Yes I made this GIF animation

**Comment by Saatvik21** [comment]
@HarshCasper I've made the required changes. Please Review. Thanks!

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**